src/bytes.rs

```diff
@@ -145,6 +145,39 @@ impl Bytes {
         }
     }
 
+    /// No cost BytesMut to Bytes converter
+    ///
+    /// # Safety
+    /// Make sure `bytes.kind()` is `KIND_VEC`
```

Comment:
can we add an assertion (or at least a debug assertion?) in this function before performing the conversion?

Reply:
I agree, but `.kind ()` is private, make it `pub (crate)`?
https://github.com/tokio-rs/bytes/blob/master/src/bytes_mut.rs#L841-L844